src/java/org/apache/cassandra/index/sai/utils/PrimaryKey.java

```diff
@@ -92,25 +93,17 @@ abstract class AbstractPrimaryKey implements PrimaryKey
         {
             public ByteSource asComparableBytes(ByteComparable.Version version)
             {
-                ByteSource tokenComparable = token().asComparableBytes(version);
-                if (partitionKey() == null)
-                    return ByteSource.withTerminator(version == ByteComparable.Version.LEGACY ? ByteSource.END_OF_STREAM
-                                                                                              : ByteSource.TERMINATOR,
-                                                     tokenComparable,
-                                                     null,
-                                                     null);
                 ByteSource keyComparable = ByteSource.of(partitionKey().getKey(), version);
+                if (clusteringComparator.size() == 0)
+                    return keyComparable;
                 // It is important that the ClusteringComparator.asBytesComparable method is used
                 // to maintain the correct clustering sort order
-                ByteSource clusteringComparable = clusteringComparator == null ||
-                                                  clusteringComparator.size() == 0 ||
-                                                  clustering() == null ||
+                ByteSource clusteringComparable = clustering() == null ||
                                                   clustering().isEmpty() ? null
```

Comment:
It's kind of funny that, because we have `@Nullable` on `clustering()`, IDEA throws up a bunch of warnings about this possibly causing an NPE...even though we have a short-circuiting `clustering() == null ||` immediately before it. We could always throw `@SuppressWarnings("DataFlowIssue")` on these methods...

Reply:
It kind of makes sense because the IDE isn't clever enough to realize that `PrimaryKey#clustering()` always returns the same value. However, it wouldn't complain if the method was named as a getter, as in `PrimaryKey#getClustering()`. I guess we can also put the clustering into a local var to prevent the warning without relying on `@SuppressWarnings`:
```java
Clustering<?> clustering = clustering();
ByteSource clusteringComparable = clustering == null || clustering.isEmpty()
                                  ? null
                                  : clusteringComparator.asByteComparable(clustering)
                                                        .asComparableBytes(version);
```
There are multiple IDE warnings around `partitionKey()` and `clustering()` being null that could be gone with whichever of these approaches we prefer (suppress warnings, rename as getters or use local vars).